vim--ai-scribe-app/src/vimOs/useUpdateEncounter.ts

```diff
@@ -1,12 +1,95 @@
-import { useMemo } from "react";
+import { useEffect, useMemo, useState } from "react";
 import type { EHR } from "vim-os-js-browser/types";
 import { useVimOsContext } from "@/providers/VimOSContext";
 
+export const useUpdateEncounterSubscription = <
+  T extends keyof EHR.UpdateEncounterParams
+>(
+  subscriptionParams: EHR.CanUpdateEncounterParams,
+  sectionName: T,
+  priorityList: Partial<Record<keyof EHR.UpdateEncounterParams, string[]>> // TODO Support Actual Fieldname // Partial<Record<T, Array<keyof EHR.UpdateEncounterParams[T]>>>
+) => {
+  const vimOS = useVimOsContext();
+  const [canUpdateSubscriptionParams, setCanUpdateSubscriptionParams] =
+    useState<boolean>(false);
+  const [subscriptionUpdatableField, setSubscriptionUpdatableField] =
+    useState<string>();
+
+  useEffect(() => {
+    const onUpdate = () => {
+      if (subscriptionParams) {
+        const { canUpdate, details } =
+          vimOS.ehr.resourceUpdater.canUpdateEncounter(subscriptionParams);
+        if (sectionName && priorityList) {
+          setCanUpdateSubscriptionParams(canUpdate);
+
+          const section = details[sectionName] || {};
+          const priorities = priorityList[sectionName] || [];
+          const updatableField = priorities?.find(
+            (fieldName) => section[fieldName as keyof typeof section] === true
+          );
+          setSubscriptionUpdatableField(updatableField);
+        }
+      }
+    };
+
+    vimOS.ehr.resourceUpdater.subscribe("encounter", onUpdate);
+
+    return () => vimOS.ehr.resourceUpdater.unsubscribe("encounter", onUpdate);
+  }, [
+    priorityList,
+    sectionName,
+    subscriptionParams,
+    vimOS.ehr.resourceUpdater,
+  ]);
+
+  return useMemo(
+    () => ({
+      canUpdateSubscriptionParams,
+      subscriptionUpdatableField,
+      updateSubscriptionField: (content: string) => {
+        if (!sectionName || !canUpdateSubscriptionParams) {
```

Comment:
I'm not sure how this works.
`canUpdate` (and `canUpdateSubscriptionParams`) will only be true if all subscriptionParams are updatable.
But then, you check which of the priorities is true; if the first one is false, `canUpdateSubscriptionParams` will also be false, and then `updateSubscriptionField` below will not work